### PR TITLE
If 193 asset series specifics data

### PR DESCRIPTION
**Pull request by seemueller**
Changes on hash method necessary to work with fieldsource sets without is's

### CODE REVIEW

**Comment by mattmikulina** [comment]
Task linked: [IF-193 specifics of data availability](https://app.clickup.com/t/2177764/IF-193)

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetSeriesService.java [hunk 1]
```diff
@@ -71,46 +70,44 @@ public AssetSeries getAssetSeriesByCompany(final Long companyId, final Long asse
                 .orElseThrow(ResourceNotFoundException::new);
     }
 
-    public AssetSeries createAssetSeriesFromAssetTypeTemplate(final Long targetCompanyId,
-                                                              final Long assetTypeTemplateId) {
+    @Transactional
```
Comment (jsylupp) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetSeriesService.java:73]: is it mandatory to set @transactional above every create method with nested objects if it already in the class declaration?

File: fusionfrontend/src/app/fleet/components/content/asset-series-create/asset-series-create.component.ts [hunk 2]
```diff
@@ -113,17 +95,29 @@ export class AssetSeriesCreateComponent implements OnInit {
     let result = true;
     switch (this.step) {
       case 1:
-        result = this.assetSeries?.name !== undefined;
+        result = this.assetSeries?.name?.length && this.assetSeries?.name?.length !== 0;
+        break;
+      case 2:
```
Comment (jsylupp) [inline fusionfrontend/src/app/fleet/components/content/asset-series-create/asset-series-create.component.ts:100]: consider using enum for values here (see ATT-wizard)

File: fusionfrontend/src/app/fleet/components/content/asset-series-create/view-mode.enum.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export enum ViewMode {
```
Comment (jsylupp) [inline fusionfrontend/src/app/fleet/components/content/asset-series-create/view-mode.enum.ts:1]: Just an refactoring idea: Maybe we could use the same enum for all Dialogs and Wizards (at least per module)

**Review by jsylupp — CHANGES_REQUESTED** [review]
Sieht gut aus!
Habe nur Kleinigkeiten angemerkt.

Melde dich einfach nochmal, wenn die Tests im Backend entsprechend korrigiert sind.

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetSeriesService.java [hunk 4]
```diff
@@ -71,46 +70,44 @@ public AssetSeries getAssetSeriesByCompany(final Long companyId, final Long asse
...
-                                .sourceUnit(fieldTarget.getField().getUnit())
-                                .build())
-                .collect(Collectors.toList());
-        List<FieldSource> savedFieldSources =
-                StreamSupport.stream(fieldSourceRepository.saveAll(newFieldSources).spliterator(), false)
-                        .collect(Collectors.toList());
-        newAssetSeries.getFieldSources().addAll(savedFieldSources);
-
-        return newAssetSeries;
+        assetSeries.setCompany(targetCompany);
```
Comment (jsylupp) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetSeriesService.java:82]: Shouldn't we add the new assetSeries to the list in company (bidirectional)? 
targetCompany.getAssetsSeries().add(assetSeries);

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetSeriesService.java [hunk 5]
```diff
@@ -71,46 +70,44 @@ public AssetSeries getAssetSeriesByCompany(final Long companyId, final Long asse
...
+        assetSeries.setAssetTypeTemplate(assetTypeTemplate);
```
Comment (jsylupp) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetSeriesService.java:83]: Same with reference in AssetTypeTemplate

**Review by jsylupp — APPROVED** [review]

**Pull request merged**